ng2-material/components/pagination/pagination.ts

```diff
@@ -0,0 +1,300 @@
+      <option *ngFor="#length of itemsPerPageOptions" [value]="length">
+        {{length}}
+      </option>
+    </select>
+    <span *ngIf="itemsPerPageAfter" class="md-pagination-items-per-page-label md-pagination-items-per-page-after">{{itemsPerPageAfter}}</span>
+  `,
+  host: {
+    '[class.md-pagination-items-per-page]': 'true',
+    '[hidden]': '!itemsPerPageOptions.length'
+  }
+})
+export class MdPaginationItemsPerPage {
+  @Input('items-per-page-before')
+  itemsPerPageBefore: string = 'Rows per page:';
+
+  @Input('items-per-page-after')
+  itemsPerPageAfter: string;
+
+  @Input('items-per-page-options')
+  itemsPerPageOptions: Array<number> = [];
+
+  @Input()
+  name: string = 'default';
+
+  @Input()
+  model: IPaginationModel = {
+    currentPage: 0,
+    itemsPerPage: 0,
+    totalItems: 0
+  };
+
+  constructor(private service: PaginationService) {
+    this.service.onChange
+      .filter(event => isPresent(event) && isPresent(event.name))
+      .filter(event => event.target === this.name)
+      .subscribe(event => {
+        if (isPresent(event.pagination)) {
+          this.model = event.pagination;
+        }
+      });
+  }
+
+  changePaginationLength(value) {
+    let model = JSON.parse(JSON.stringify(this.model));
+    model.currentPage = 1;
+    model.itemsPerPage = parseInt(value);
+    this.service.change(this.name, model);
+  }
+
+}
+
+export interface IPaginationChange {
+  name: string;
+  target: string;
+  pagination: IPaginationModel;
+}
+
+@Component({
+  selector: 'md-pagination',
+  template: `
+    <ng-content></ng-content>
+    <div #default></div>
+  `,
+  directives: [MdPaginationRange, MdPaginationControls, MdPaginationItemsPerPage],
+  host: {
+    '[class.md-pagination]': 'true'
+  }
+})
+export class MdPagination implements AfterContentInit, AfterViewInit {
+  @Input()
+  name: string = 'default';
+
+  @Input()
+  model: IPaginationModel;
+
+  @Input()
+  range: boolean = true;
+
+  @Input('range-format')
+  rangeFormat: string;
+
+  @Input()
+  controls: boolean = true;
+
+  @Input('items-per-page')
+  itemsPerPage: boolean = true;
+
+  @Input('items-per-page-before')
+  itemsPerPageBefore: string;
+
+  @Input('items-per-page-after')
+  itemsPerPageAfter: string;
+
+  @Input('items-per-page-options')
+  itemsPerPageOptions: Array<number>;
+
+  @Output('on-pagination-change')
+  onPaginationChange: EventEmitter<IPaginationChange> = new EventEmitter(false);
+
+  constructor(private service: PaginationService, private element: ElementRef, private componentLoader: DynamicComponentLoader) {
+    this.service.onChange
+      .filter(event => isPresent(event) && isPresent(event.name))
+      .filter(event => event.target === this.name)
+      .subscribe(event => this.onPaginationChange.emit(event));
+
+  }
+
+  public initDefaultLayout() {
+    if (this.itemsPerPage === true) {
```

Comment:
Why do you need to use DCL here? Couldn't you just have all the components in the template with `*ngIf` statements?

Maybe I'm missing something?